public/app/plugins/datasource/loki/components/LokiQueryEditorByApp.test.tsx

```diff
@@ -40,24 +41,23 @@ describe('LokiQueryEditorByApp', () => {
     expect(queryByTestId(regularTestIds.editor)).toBeNull();
   });
 
-  it('should render regular query editor for unkown apps', () => {
+  it('should render regular query editor for unknown apps', async () => {
     const { getByTestId, queryByTestId } = setup(CoreApp.Unknown);
-
-    expect(getByTestId(regularTestIds.editor)).toBeInTheDocument();
+    expect(await waitFor(() => getByTestId(regularTestIds.editor))).toBeInTheDocument();
     expect(queryByTestId(alertingTestIds.editor)).toBeNull();
   });
 
-  it('should render expore query editor for explore', () => {
+  it('should render regular query editor for explore', async () => {
     const { getByTestId, queryByTestId } = setup(CoreApp.Explore);
 
-    expect(getByTestId(exploreTestIds.editor)).toBeInTheDocument();
+    expect(await waitFor(() => getByTestId(regularTestIds.editor))).toBeInTheDocument();
     expect(queryByTestId(alertingTestIds.editor)).toBeNull();
   });
 
-  it('should render regular query editor for dashboard', () => {
+  it('should render regular query editor for dashboard', async () => {
     const { getByTestId, queryByTestId } = setup(CoreApp.Dashboard);
 
-    expect(getByTestId(regularTestIds.editor)).toBeInTheDocument();
+    expect(await waitFor(() => getByTestId(regularTestIds.editor))).toBeInTheDocument();
```

Comment:
You can consider:

```suggestion
    expect(await screen.findByTestId(regularTestIds.editor)).toBeInTheDocument();
```

https://testing-library.com/docs/example-findbytext/

Reply:
Yes I agree! It is also mentioned in one of my favourite and most useful blog posts https://kentcdodds.com/blog/common-mistakes-with-react-testing-library.